common/src/test/java/io/netty/util/internal/logging/Log4J2LoggerTest.java

```diff
@@ -15,211 +15,82 @@
+ * so There is no need to test it's method.<br>
+ * We only need to test the netty's {@linkplain InternalLogger} interface method.<br>
+ * It's meaning that we only need to test the Override method in the {@linkplain Log4J2Logger}.
+ */
+public class Log4J2LoggerTest extends AbstractInternalLoggerTest<Logger> {
+
+    {
+        mockLog = LogManager.getLogger(loggerName);
+        logger = new Log4J2Logger(mockLog) {
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public void logMessage(String fqcn, Level level, Marker marker, Message message, Throwable t) {
+                result.put("level", level.name());
+                result.put("t", t);
+                super.logMessage(fqcn, level, marker, message, t);
+            };
+        };
     }
 
     @Test
-    public void testWarn() {
-        Logger mock = mock(Logger.class);
-
-        when(mock.getName()).thenReturn("foo");
-
-        InternalLogger logger = new Log4J2Logger(mock);
-        logger.warn("a");
-
-        verify(mock).getName();
-        verify(mock).warn("a");
+    public void testEXCEPTION_MESSAGE() {
+        assertEquals(getFieldValue(AbstractInternalLogger.class, "EXCEPTION_MESSAGE"),
+                getFieldValue(Log4J2Logger.class, "EXCEPTION_MESSAGE"));
     }
 
-    @Test
-    public void testWarnWithException() {
-        Logger mock = mock(Logger.class);
-
-        when(mock.getName()).thenReturn("foo");
-
-        InternalLogger logger = new Log4J2Logger(mock);
-        logger.warn("a", e);
-
-        verify(mock).getName();
-        verify(mock).warn("a", e);
+    @SuppressWarnings("unchecked")
+    private static <T> T getFieldValue(Class<?> clazz, String fieldName) {
+        try {
+            Field field = clazz.getDeclaredField(fieldName);
+            if (!field.isAccessible()) {
+                field.setAccessible(true);
+            }
+            return (T) field.get(AbstractInternalLogger.class);
+        } catch (ReflectiveOperationException e) {
+            throw new RuntimeException(e);
+        }
     }
 
-    @Test
-    public void testError() {
-        Logger mock = mock(Logger.class);
-
-        when(mock.getName()).thenReturn("foo");
-
-        InternalLogger logger = new Log4J2Logger(mock);
-        logger.error("a");
-
-        verify(mock).getName();
-        verify(mock).error("a");
+    @Override
+    protected void setLevelEnable(InternalLogLevel level, boolean enable) throws Exception {
+        Level targetLevel = Level.valueOf(level.name());
+        if (!enable) {
+            Level[] levels = Level.values();
+            Arrays.sort(levels);
+            int pos = Arrays.binarySearch(levels, targetLevel);
+            targetLevel = levels[pos - 1];
+        }
+
+        Method method = mockLog.getClass().getDeclaredMethod("setLevel", Level.class);
+        if (!method.isAccessible()) {
+            method.setAccessible(true);
```

Comment:
Use `ReflectionUtil.trySetAccessible(...)` like here:

https://github.com/netty/netty/commit/5c572f0f63da22392d477697c4c289072915d483#diff-0ad6a10cb2bef26affd37693931ef838R661

Reply:
I think this is no problem, if throw exception just say the test failed.
but I fix use `ReflectionUtil.trySetAccessible(...)`.
Thank you, let me see the accessible different in jdk9 : )